cmd/sanitize-prow-jobs/main.go

```diff
@@ -59,6 +59,11 @@ func determinizeJobs(prowJobConfigDir string, config *dispatcher.Config) error {
 			return nil
 		}
 
+		matches := config.AllMatchingPathRegex(path)
+		if len(matches) > 1 {
+			errChan <- fmt.Errorf("path %s matches more than 1 regex: %s", path, matches)
+		}
+
```

Comment:
It's better, but I think this should jsut be somewhere under `GetClusterForJob`. Somewhere under that method, we end up doing the same thing (iterate over regexes), and that method should IMO do the check itself, not rely on some code doing it first.

The `DetermineClusterForJob` method seems like a good place. That method should not rely on the config being pre-checked, it should detect the possible duplicity and end with an error (which needs to be propagate ot the `defaultJobConfig` here.

Reply:
Done.

Propagate the error all the way up.